api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -210,36 +211,56 @@ private void updateUserAccessTiers(DbUser dbUser, Agent agent) {
     tiersForRemoval.forEach(tier -> accessTierService.removeUserFromTier(dbUser, tier));
   }
 
-  private boolean shouldUserBeRegistered(DbUser user) {
-    boolean dataUseAgreementCompliant = false;
+  public boolean isCompleteAndNotExpired(Timestamp completionTime) {
+    Timestamp expirationTime =
+        new Timestamp(
+            clock.millis()
+                - TimeUnit.MILLISECONDS.convert(
+                    configProvider.get().accessRenewal.expiryDays, TimeUnit.DAYS));
+    if (configProvider.get().access.enableAccessRenewal) {
+      return completionTime != null && expirationTime.before(completionTime);
+    }
+    return completionTime != null;
+  }
+
+  private boolean isDataUseAgreementCompliant(DbUser user) {
     if (user.getDataUseAgreementBypassTime() != null
         || !configProvider.get().access.enableDataUseAgreement) {
       // Data use agreement version may be ignored, since it's bypassed on the user or env level.
-      dataUseAgreementCompliant = true;
+      return true;
     } else if (user.getDataUseAgreementSignedVersion() != null
-        && user.getDataUseAgreementSignedVersion() == getCurrentDuccVersion()) {
+        && user.getDataUseAgreementSignedVersion() == getCurrentDuccVersion()
+        && isCompleteAndNotExpired(user.getDataUseAgreementCompletionTime())) {
       // User has signed the most-recent DUCC version.
-      dataUseAgreementCompliant = true;
+      return true;
     }
-    boolean eraCommonsCompliant =
-        user.getEraCommonsBypassTime() != null
-            || !configProvider.get().access.enableEraCommons
-            || user.getEraCommonsCompletionTime() != null;
-    boolean complianceTrainingCompliant =
-        user.getComplianceTrainingCompletionTime() != null
-            || user.getComplianceTrainingBypassTime() != null
-            || !configProvider.get().access.enableComplianceTraining;
+    return false;
+  }
+
+  private boolean isEraCommonsCompliant(DbUser user) {
```

Comment:
nit: could you add a comment explaining that ERA Commons is not subject to annual renewal, and therefore we don't need to check its expiration?  (also for other modules like this)